lib/Telemetry/Monitor.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * @author Thomas Müller <thomas.mueller@tmit.eu>
+ *
+ * @copyright Copyright (c) 2018, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\ConfigReport\Telemetry;
+
+use OC\License\LicenseFetcher;
+use OCA\ConfigReport\ReportDataCollector;
+use OCP\Http\Client\IClientService;
+use OCP\IConfig;
+use OCP\ILogger;
+
+class Monitor {
+	private IConfig $config;
+	private ILogger $logger;
+	private IClientService $clientService;
+	private ReportDataCollector $collector;
+
+	public function __construct(IConfig $config, ILogger $logger, IClientService $clientService, ReportDataCollector $collector) {
+		$this->config = $config;
+		$this->logger = $logger;
+		$this->clientService = $clientService;
+		$this->collector = $collector;
+	}
+
+	public function sendTelemetry(): void {
+		$telemetryOptIn = $this->config->getSystemValue('telemetry.enabled', true);
+		if (!$telemetryOptIn) {
+			return;
+		}
+
+		$data = $this->collectTelemetryData();
+
+		$client = $this->clientService->newClient();
+		try {
+			$client->post('https://3qyh826rmd.execute-api.ap-southeast-1.amazonaws.com/default/pocTelemetry', [
+				'body' => json_encode($data, JSON_THROW_ON_ERROR),
+			]);
+		} catch (\Exception $e) {
+			$this->logger->logException($e);
+		}
+	}
+
+	private function getLicenseKey(): string {
+		try {
+			$lf = \OC::$server->query(LicenseFetcher::class);
+			if ($lf instanceof LicenseFetcher) {
+				$l = $lf->getOwncloudLicense();
+				if ($l) {
+					return $l->getLicenseString();
+				}
+			}
+		} catch (\Exception $e) {
+			$this->logger->logException($e);
+		}
+
+		return "";
+	}
+
+	private function collectTelemetryData(): array {
+		$licenseKey = $this->getLicenseKey();
```

Comment:
Should we check for the license state? (https://github.com/owncloud/core/blob/master/lib/public/License/ILicenseManager.php#L95)
We might not be interested in invalid or expired licenses.

Reply:
see above - it is used as an identifier.

Reply:
The license can be changed anytime, specially if it's about to expire. Any instance could have multiple licenses during its lifetime (one at a time). Using the license as an identifier doesn't seem good.
In addition, I think we should consider the license as a password in regards of confidentiality. I mean, even if leaking the license might be harmless for the instance, there could be other instances taking advantage of the leaked license. If we need to send it, we probably should either hash it or encrypt it.

Reply:
All known and understood. The current implementation was discussed and agreed on offline with different parties.
Nothing we need to discuss in here again. THX a lot for your comments!